### PR TITLE
Integration: profile edit #36

### LINKED ISSUE #23: FE: Profile Settings > Edit Profile
Create the UI for specifically "edit profile" content

- refer to `setup-profile-edit.png` in provided assets
- This task is not responsible for navbar

### LINKED ISSUE #36: IN: Profile Page > Edit Profile
Integrate the Profile page. On save and update, this should update the profile on the backend

### PR DESCRIPTION
**Pull request by dTechLife**
### What this PR does (required):
~This is a draft
-Fully integrate creating of new profile pages and editing of them after creation. 
### Screenshots / Videos (front-end only):
~This is a draft
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

### Any information needed to test this feature (required):
A few more touchups are required for this integration, it's waiting on another feature as well. 
Check back soon. 

### Any issues with the current functionality (optional):
- Edit profile still needs to be tested and perfected. (put) 
-A lot still!

### CODE REVIEW

File: client/src/pages/Dashboard/Dashboard.tsx
Comment (ezeilo-su) [inline client/src/pages/Dashboard/Dashboard.tsx:26]: We need to remove this console log and add the actual logic.

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:7]: Remove this one too - :smile:

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:39]: I'd suggest you make proper use of `async/await` to simplify your code and make it more readable.

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:40]: This console log is not needed. So let's remove it.

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:71]: Also remove this console log, please

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:89]: This one too

File: client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx
Comment (ezeilo-su) [inline client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx:92]: We don't need this alert - :smile:

File: server/controllers/profile.js
Comment (ezeilo-su) [inline server/controllers/profile.js:21]: :+1:

File: server/routes/profile.js
Comment (ezeilo-su) [inline server/routes/profile.js:11]: For the `update` route, you could make a dynamic route `router.route("/update/:id")` so that `id` is automatically available in the `req.params` object

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
This is a remarkable work on this one - :+1: 
Just check through the comments and see where to make changes. It's also a good idea to provide a few screenshots of your results in your PRs.
Happy coding!

File: server/routes/profile.js
Comment (DestinTech) [reply server/routes/profile.js:11]: I'm fairly certain that req.user.id is populated with the users data, I used req.user.id as this prevents unauthorized modifications I believe. Let me know what you think about that, thanks!

Comment (ezeilo-su) [reply server/routes/profile.js:11]: OK, though I can recall you used something like this: `req.params.id || req.user.id`. Which one works for you is fine :smiley:

**Comment by ezeilo-su** [comment]
Please, also resolve this merge conflict.

**Review by ugur-akin — COMMENTED** [review]
This PR shows some old changes, I believe it's because git doesn't track changes to the target branch (e.g. we merged things into develop after we opened this PR). This [thread](https://stackoverflow.com/q/16306012) is a bit old but gives more detail about the issue, and it looks to me like [this](https://stackoverflow.com/a/26986320) is the simplest way to work around it. 

Can you update the PR so we can go over it one last time and merge it soon?

File: server/routes/profile.js
Comment (ezeilo-su) [inline server/routes/profile.js:6]: You can organise your files so: `'../controllers/profile/update.js'`. Just a suggestion, though.

File: server/routes/profile.js
Comment (ezeilo-su) [inline server/routes/profile.js:11]: Add the `id` of the resource you intend to update in the request params `'/:id'`

File: server/controllers/profile.js
Comment (ezeilo-su) [inline server/controllers/profile.js:80]: Let's remove this comment, please.

File: server/controllers/profile.js
Comment (ezeilo-su) [inline server/controllers/profile.js:77]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
Good work making those changes :+1:

File: server/routes/profile.js
Comment (DestinTech) [reply server/routes/profile.js:6]: This sounds like a great idea, I'd need to integrate a few other branches others have made with this change though. If we stumble upon a need we will reformat the code with this in mind and push a hotfix. Thank you!

**Comment by rajivtitus** [comment]
@DestinTech No problem! I just wanted to check with you as I did not have a chance to review this again. In the future please wait for approval prior to merging PRs :)

**Pull request merged**
